src/App.js

```diff
@@ -1,25 +1,69 @@
-import logo from './logo.svg';
+// src/App.js
+import { useState } from 'react';
 import './App.css';
+import foodsDataJSON from './foods.json';
+import FoodBox from './components/FoodBox';
+import AddFoodForm from './components/AddFoodForm';
+import SearchFood from './components/SearchFood';
+import { Input, Col, Divider, Button, Row } from 'antd';
 
 function App() {
+  const [allFoods, setAllFoods] = useState(foodsDataJSON);
+  const [foods, setFoods] = useState(foodsDataJSON);
+  const [isShowing, setIsShowing] = useState(true);
+
+  const addNewFood = (foodObj) => {
+    const updatedFoods = [foodObj, ...foods];
+    const updatedAllFoods = [foodObj, ...allFoods];
+    console.log('foodObj :>> ', foodObj);
+    setFoods(updatedFoods);
+    setAllFoods(updatedAllFoods);
+  };
+
+  const searchFoodList = (char) => {
+    let filteredFoods;
+    if (char === ' ') {
+      filteredFoods = allFoods;
+    } else {
+      filteredFoods = foods.filter((oneFood) => {
+        return oneFood.name.toLowerCase().includes(char.toLowerCase());
+      });
+    }
+    console.log('filteredFoods :>> ', filteredFoods);
+
+    setAllFoods(filteredFoods);
+  };
```

Comment:
The search was not working properly, again. We need to swap what's being filtered and what's being updated, in the state.

Also, you don't need the if-else statement, this works, as is.

```suggestion
  const searchFoodList = (char) => {
  
  const filteredFoods = allFoods.filter((oneFood) => {
      return oneFood.name.toLowerCase().includes(char.toLowerCase());
    });

    setFoods(filteredFoods);
  };
```

Reply:
I think the bugs are fixed. =)
Thank you, Pedro!